src/Uno.UI/UI/Xaml/DependencyObjectStore.Binder.cs

```diff
@@ -105,10 +105,21 @@ void SetInherited(IDependencyObjectStoreProvider provider)
 					provider.Store.SetInheritedDataContext(inheritedValue);
 				}
 			}
-
 			for (int i = 0; i < _childrenBindable.Count; i++)
 			{
 				var child = _childrenBindable[i];
+				var parent = child?.GetParent();
+
+				//Do not propagate value if you are not this child's parent
+				//Covers case where a child may hold a binding to a view higher up the tree
+				//Example: Button A contains a Flyout with Button B inside of it
+				//	Button B has a binding to the Flyout itself
+				//	We should not propagate Button B's DataContext to the Flyout
+				//	since its real parent is actually Button A 
+				if (parent != null && parent != ActualInstance)
```

Comment:
Are we sure that we won't leak the `DataContext` when we remove a child from it's  parent and we pout it in a pool?

Even if I dislike it a lot, the way uno is avoiding datacontext leaks is by explicitly pushing a `null` as `DataContext` when an element is unloaded. If here we stop that propagation when the parent is not set, I would make sure that we still are allowing that `null` propagation (i.e. the parent is removed after the `null` has been pushed).

Reply:
If parent is not set, it'll be `null`, which is being checked for. It's only if parent is non-null **and** something else that we skip propagating the DataContext.